array.h

```diff
@@ -2746,6 +2759,37 @@ const_array_ref<T, NewShape> reinterpret_shape(
   return reinterpret_shape(a.cref(), new_shape, offset);
 }
 
+/** Move an array `from` to a new array, reinterpreting the shape of the array
+ * to `new_shape`, with a base pointer offset `offset`. This is only available
+ * for trivial `T`. */
+template <typename NewShape, typename T, typename Shape, typename Alloc>
+array<T, NewShape, Alloc> move_reinterpret_shape(array<T, Shape, Alloc>&& from, const NewShape& new_shape, index_t offset = 0) {
+  // TODO: Use enable_if to implement this check. It is difficult to do due to
```

Comment:
I think if constexpr is equivalent to static_assert in the way that I care about, which is the advantage of enable_if is that it generates an error in the user code instead of in array.h.